### ISSUE TITLE
S3 Source custom partition Caused by: NoSuchMethodException: after creation 

### ISSUE THREAD

**Issue by artemio77** [confluentinc/kafka-connect-storage-common#239]
Hi, I implement a custom partitioner for the sink and source s3 connector.

**Comment by yarlagaddag** [comment]
@artemio77 is this issue resolved? I am also getting the same problem no matter what partitioner I use i.e default, or timebased or daily

any pointers would be helpful

**Comment by OneCricketeer** [comment]
This connector doesn't have field and time partitioner. Therefore, it wouldn't be found, by default. 

Regarding the original post, seems like it was working? 

> When I create sink s3 connector everuthing is fine it working correct 

> ... `com.kafka.connect.extention.FieldAndTimeBasedPartitioner`

Not clear what is this class since Kafka Connect packages start with `org.apache.kafka`

**Comment by yarlagaddag** [comment]
@OneCricketeer what are the valid partitioners for this connector?

**Comment by OneCricketeer** [comment]
@yarlagaddag They are all here, but switch the branch/tag to the version you've installed https://github.com/confluentinc/kafka-connect-storage-common/tree/master/partitioner/src/main/java/io/confluent/connect/storage/partitioner

**Comment by yarlagaddag** [comment]
Thank you @OneCricketeer . So there is TimeBasedPartitioner, and this is what I used earlier after looking into the above

**Comment by yarlagaddag** [comment]
Got the same issue for DefaultPartitioner as well
org.apache.kafka.common.config.ConfigException: Invalid value java.lang.NoSuchMethodException: io.confluent.connect.storage.partitioner.DefaultPartitioner.<init>(io.confluent.connect.cloud.storage.source.StorageSourceConnectorConfig, io.confluent.connect.cloud.storage.source.SourceStorage) for configuration Failed to instantiate partitioner class io.confluent.connect.storage.partitioner.DefaultPartitioner\n\tat io.confluent.connect.cloud.storage.source.StorageSourceConnectorConfig.getPartitioner(StorageSourceConnectorConfig.java:686)\n\tat io.confluent.connect.cloud.storage.source.StorageSourceTask.start(StorageSourceTask.java:75)\n\tat org.apache.kafka.connect.runtime.WorkerSourceTask.initializeAndStart(WorkerSourceTask.java:230)\n\tat org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:198)\n\tat org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:255)\n\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\n"
    },

**Comment by OneCricketeer** [comment]
Right. If you want to use both fields and timestamp, you'll need to checkout PR #251, then replace all the connect-storage-common JARs on your connect servers

**Comment by yarlagaddag** [comment]
alright.. I am using just the default one and still ran into the same issue, not sure what else is causing this issue.. stack trace is posted above

**Comment by OneCricketeer** [comment]
I'd guess it has something to do with reflection looking up the class name, then calling configure method since all the partitioners themselves have been tested since ac6daadfb4d49b02def0cac3b55b5baf3d866c8a

**Comment by yarlagaddag** [comment]
Hmm, ok. thanks @OneCricketeer
